### PR TITLE
cache-manager

### PR DESCRIPTION
**Pull request by marudor**

### CODE REVIEW

File: definitions/npm/cache-manager_v2.x.x/flow_>=v0.28.x/cache-manager_v2.x.x.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
+declare module 'cache-manager' {
+  declare type _CacheOptions = {
+    ttl?: number,
```
Comment (thejameskyle) [inline definitions/npm/cache-manager_v2.x.x/flow_>=v0.28.x/cache-manager_v2.x.x.js]: What about `max`, `store`, `isCacheableValue`?

Looking at the source there is also `ignoreCacheErrors` and `promiseDependency`

Comment (thejameskyle) [inline definitions/npm/cache-manager_v2.x.x/flow_>=v0.28.x/cache-manager_v2.x.x.js]: Oh sorry this was meant for `caching(opts)`

Comment (marudor) [inline definitions/npm/cache-manager_v2.x.x/flow_>=v0.28.x/cache-manager_v2.x.x.js]: Thanks, This was copied straight from out private work libdefs.
Wasn't surprised something was missing.

Added options & wrap function.

File: definitions/npm/cache-manager_v2.x.x/flow_>=v0.28.x/cache-manager_v2.x.x.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+  }
+  declare type CacheOptions = $Shape<_CacheOptions> | Object
```
Comment (jeffmo) [inline definitions/npm/cache-manager_v2.x.x/flow_>=v0.28.x/cache-manager_v2.x.x.js]: `$Shape<_CacheOptions> | Object` is actually just as permissive/restrictive as just `Object`: "Either an object with this shape, or any object" is the same as "any object".

We don't really have a good way of saying what I think you're trying to say here.

Comment (marudor) [inline definitions/npm/cache-manager_v2.x.x/flow_>=v0.28.x/cache-manager_v2.x.x.js]: Nope there is not.
All extra Settings get passed to the specific Store implementation. So we can't restrict it to _CacheOptions.

**Pull request merged**